includes/class-webmention-receiver.php

```diff
@@ -465,49 +455,144 @@ public static function webmention_verify( $data ) {
 			);
 		}
 
-		// A valid response code from the other server would not be considered an error.
-		$response_code = wp_remote_retrieve_response_code( $response );
-		// not an (x)html, sgml, or xml page, no use going further
-		if ( preg_match( '#(image|audio|video|model)/#is', wp_remote_retrieve_header( $response, 'content-type' ) ) ) {
+		$response = self::head( $data['source'] );
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+		$response = self::fetch( $data['source'] );
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$remote_source_original = wp_remote_retrieve_body( $response );
+
+		// check if source really links to target
+		if ( ! strpos(
+			htmlspecialchars_decode( $remote_source_original ),
+			str_replace(
+				array(
+					'http://www.',
+					'http://',
+					'https://www.',
+					'https://',
+				),
+				'',
+				untrailingslashit( preg_replace( '/#.*/', '', $data['target'] ) )
+			)
+		) ) {
 			return new WP_Error(
-				'unsupported_content_type',
-				esc_html__( 'Content Type is not supported', 'webmention' ),
+				'target_not_found',
+				esc_html__( 'Cannot find target link', 'webmention' ),
 				array(
 					'status' => 400,
 					'data'   => $data,
 				)
 			);
 		}
 
-		switch ( $response_code ) {
+		if ( ! function_exists( 'wp_kses_post' ) ) {
+			include_once ABSPATH . 'wp-includes/kses.php';
+		}
+
+		$remote_source = wp_kses_post( $remote_source_original );
+		$content_type  = self::get_content_type( $response );
+		$commentdata   = compact( 'remote_source', 'remote_source_original', 'content_type' );
+
+		return array_merge( $commentdata, $data );
+	}
+
+	/**
+	 * Return Arguments for Retrieving Webmention URLs
+	 *
+	 * @return array
+	 */
+	public static function remote_arguments() {
+
+		$wp_version = get_bloginfo( 'version' );
+		$user_agent = apply_filters( 'http_headers_useragent', 'WordPress/' . $wp_version . '; ' . get_bloginfo( 'url' ) );
+		$args       = array(
+			'timeout'             => 100,
+			'limit_response_size' => 153600,
+			'redirection'         => 20,
+			'headers'             => array(
+				'Accept' => 'application/mf2+json, text/html, text/plain',
```

Comment:
I would love to add `application/mf2+json` as a "beta" opt-in setting. I think, for now, the html representations are much more detailed than any `mf2` feeds.

Reply:
This was based on the fact Semantic Linkbacks used mf2+json.  But if you look at the priority, it will only offer it if the endpoint returns it. So far, only @aaronpk does as far as I know, so not an issue to drop it for now.